applications/RANSApplication/tests/cpp/evm_k_epsilon/test_k_epsilon_utilities.cpp

```diff
@@ -0,0 +1,406 @@
+#include "custom_utilities/test_utilities.h"
+#include "rans_application_variables.h"
+
+#include "custom_processes/auxiliary_processes/rans_nut_k_epsilon_high_re_calculation_process.h"
+#include "custom_processes/auxiliary_processes/rans_nut_k_epsilon_high_re_sensitivities_process.h"
+
+namespace Kratos
+{
+namespace Testing
+{
+typedef ModelPart::NodeType NodeType;
+
+typedef ModelPart::ElementType ElementType;
+
+typedef Geometry<NodeType> GeometryType;
+
+typedef GeometryType::ShapeFunctionsGradientsType ShapeFunctionDerivativesArrayType;
+
+/**
+ * Auxiliar function to generate a triangular element to be tested.
+ */
+namespace RansEvmKEpsilonModel
+{
+void AddVariablesToModelPart(ModelPart& rModelPart)
+{
+    // Set buffer size
+    rModelPart.SetBufferSize(2);
+
+    // Variables addition
+    rModelPart.AddNodalSolutionStepVariable(DISTANCE);
+    rModelPart.AddNodalSolutionStepVariable(PRESSURE);
+    rModelPart.AddNodalSolutionStepVariable(EXTERNAL_PRESSURE);
+    rModelPart.AddNodalSolutionStepVariable(VELOCITY);
+    rModelPart.AddNodalSolutionStepVariable(MESH_VELOCITY);
+    rModelPart.AddNodalSolutionStepVariable(DENSITY);
+    rModelPart.AddNodalSolutionStepVariable(VISCOSITY);
+    rModelPart.AddNodalSolutionStepVariable(ACCELERATION);
+    rModelPart.AddNodalSolutionStepVariable(RELAXED_ACCELERATION);
+    rModelPart.AddNodalSolutionStepVariable(BODY_FORCE);
+    rModelPart.AddNodalSolutionStepVariable(KINEMATIC_VISCOSITY);
+    rModelPart.AddNodalSolutionStepVariable(TURBULENT_VISCOSITY);
+    rModelPart.AddNodalSolutionStepVariable(TURBULENT_KINETIC_ENERGY);
+    rModelPart.AddNodalSolutionStepVariable(TURBULENT_ENERGY_DISSIPATION_RATE);
+    rModelPart.AddNodalSolutionStepVariable(RANS_Y_PLUS);
+    rModelPart.AddNodalSolutionStepVariable(RANS_AUXILIARY_VARIABLE_1);
+    rModelPart.AddNodalSolutionStepVariable(RANS_AUXILIARY_VARIABLE_2);
+    rModelPart.AddNodalSolutionStepVariable(TURBULENT_KINETIC_ENERGY_RATE);
+    rModelPart.AddNodalSolutionStepVariable(TURBULENT_ENERGY_DISSIPATION_RATE_2);
+    rModelPart.AddNodalSolutionStepVariable(RANS_SCALAR_1_ADJOINT_1);
+    rModelPart.AddNodalSolutionStepVariable(RANS_SCALAR_2_ADJOINT_1);
+    rModelPart.AddNodalSolutionStepVariable(RANS_SCALAR_1_ADJOINT_2);
+    rModelPart.AddNodalSolutionStepVariable(RANS_SCALAR_2_ADJOINT_2);
+    rModelPart.AddNodalSolutionStepVariable(RANS_SCALAR_1_ADJOINT_3);
+    rModelPart.AddNodalSolutionStepVariable(RANS_SCALAR_2_ADJOINT_3);
+    rModelPart.AddNodalSolutionStepVariable(ADJOINT_FLUID_VECTOR_1);
+    rModelPart.AddNodalSolutionStepVariable(ADJOINT_FLUID_VECTOR_2);
+    rModelPart.AddNodalSolutionStepVariable(ADJOINT_FLUID_VECTOR_3);
+    rModelPart.AddNodalSolutionStepVariable(ADJOINT_FLUID_SCALAR_1);
+    rModelPart.AddNodalSolutionStepVariable(AUX_ADJOINT_FLUID_VECTOR_1);
+    rModelPart.AddNodalSolutionStepVariable(RANS_AUX_ADJOINT_SCALAR_1);
+    rModelPart.AddNodalSolutionStepVariable(RANS_AUX_ADJOINT_SCALAR_2);
```

Comment:
I mean, do all of those have to be historical? this is quite a big list!

Reply:
The adjoint bossak/steady scheme requires all of them to be historical. Since this is only a 3 node modelpart, does it affect a lot for test execution memory and time?